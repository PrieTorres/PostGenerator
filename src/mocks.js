export const mockArrayPosts = [
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 0,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 1,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 2,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 3,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 4,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 5,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 6,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  },
  {
    title: "Lorem Ipsum sit amet",
    body: "Se res de kels evamos mas durn kopa qiue soia postroer di lammet",
    id: 7,
    photo: {
      title: "doggo",
      src:{
        portrait: "https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
      }
    }
  }
];