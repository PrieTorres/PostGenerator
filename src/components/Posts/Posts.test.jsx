import { render, screen } from "@testing-library/react"
import { Posts } from "."
import { mockArrayPosts } from "../../mocks"
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';


const posts = mockArrayPosts;
describe("<Posts />", () => {
  it("should render 8 posts when it's passed an array with 8 posts", () => {
    render(<Posts posts={posts} />);
    const renderedPosts = screen.getAllByTestId("loaded-post");

    expect(renderedPosts.length).toBe(8);
  });
  it("should return text 'No posts... ;(' when get no data", () => {
    const {container} = render(<Posts posts={[]} />);
    const sadFaceTest = container.getElementsByClassName("no-posts").item(0);

    expect(sadFaceTest).toBeInTheDocument();
  });
});
