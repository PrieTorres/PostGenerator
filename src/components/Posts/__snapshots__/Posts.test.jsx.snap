// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet quit
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet quit
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="post"
    data-testid="loaded-post"
  >
    <img
      alt="doggo"
      class="post-image "
      src="https://images.pexels.com/photos/3090875/pexels-photo-3090875.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=1200&w=800"
    />
    <h2
      class="post-title"
    >
      Lorem Ipsum sit amet
    </h2>
    <p
      class="post-text"
    >
      Se res de kels evamos mas durn kopa qiue soia postroer di lammet
    </p>
  </div>
  <div
    class="no-posts"
  />
</div>
`;
